Backend/BackendAPI/Models/Bike.cs

```diff
@@ -16,5 +16,18 @@ public class Bike
         //Właściwości potrzebne do utrzymania relacji między tabelami
         public BikeStation BikeStation { get; set; }
         public IList<Rental> Rentals { get; set; }
+
+
+        //Tutaj proszę o sprawdzenie, czy dobrze zrozumiałem ideę Update
+        public Bike Modify(Bike other)
+        {
+            ID = other.ID;
+            BikeStationID = other.BikeStationID;
+            State = other.State;
+            BikeStation = other.BikeStation;
+            Rentals = new List<Rental>(other.Rentals);
+
+            return this;
```

Comment:
Tu jest bardzo niebezpieczny kod. Zmieniając ID roweru możemy popsuć relacje z bazy danych, tj. niektóre wypożyczenia mają ID roweru i my to ID zmienimy no i wtedy wyskoczy wyjątek, bo w tych wypożyczeniach klucz obcy będzie wskazywał na nieistniejący rower. Poza tym, na przyzsłość, tworząc jakikolwiek obiekt do bazy danych pola związane z relacjami za wyjątkiem kluczy obcych (BikeStationID, BikeID, CośTamID) zostawiamy jako null, te pola będą wypełnioane przez metodę Include w zapytaniach LINQ. Czyli Rentals i BikeStation powinno zostać nullem najlepiej. Osobiście, ja bym wywalił stąd metodę Modify i logikę przepisania BikeStationID,, State do metody Update z repository. W ogóle mi się nie podoba że metoda Update w BikeReposirory przyjmuje cały Bike, skoro przecież wiadomo, że zupdatowany zostaje tylko BikeStationID oraz State, no ale tak jest w specyfikacji więc tak trzeba zrobić.

Reply:
Usunąłem przepisanie obiektów logicznych oraz dodałem test.